core/src/main/java/hudson/model/Label.java

```diff
@@ -497,7 +497,7 @@ public Label paren() {
 
 
     @Override
-    public boolean equals(Object that) {
+    public final boolean equals(Object that) {
```

Comment:
If you have made equals final then you also need to make hashCode and compareTo final